kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/ConversationUIService.java

```diff
@@ -744,16 +746,28 @@ public void checkForStartNewConversation(Intent intent) {
             baseContactService.upsert(contact);
             new UserClientService(fragmentActivity).updateUserDisplayName(userId, fullName);
         }
-        String messageJson = intent.getStringExtra(MobiComKitConstants.MESSAGE_JSON_INTENT);
-        if (!TextUtils.isEmpty(messageJson)) {
-            Message message = (Message) GsonUtils.getObjectFromJson(messageJson, Message.class);
+        Message message = null;
+        if(channelKey != -1) {
+            List<Message> messages = new MessageDatabaseService(fragmentActivity).getLatestMessageByChannelKey(channelKey);
```

Comment:
Why are we fetching the list of messages and not just one message?

Reply:
I have decreased its priority at first it will fetch by json . It will be the end solution